### DISCUSSION TITLE
Test does not timeout

### DISCUSSION THREAD

**Discussion by Joost-SF** [cypress-io/cypress#15979] [Questions and Help] [answered]
I've been creating tests for the saucelabs website in my Cypress practice project, where they have one account configured which takes a very long time to load. I've created a test which I want to fail just for the sake of it failing because of a timeout.

Right now it never fails, even though the page takes a few seconds to load, the test still passes. I've tried this the following ways:
        cy.url( { timeout: 500 }).should('eq', 'https://www.saucedemo.com/inventory.html')
        cy.xpath('//span[@class="title" and contains(text(),"Products")]').should('be.visible', { timeout: 500});
        cy.xpath('//span[@class="title" and contains(text(),"Products")]', { timeout: 500}).should('be.visible');

These might be very wrong, but could someone help me out with this? It does not matter what I do, but the test passes every time even though I see that the pages takes longer than 500ms to load.

Thanks in any case!

Ps: I've also used the regular cy.get('.title', { timeout: 500}) etc.

**Comment by bahmutov** [comment, marked as answer]
Note that you _never_ change the timeout inside the assertion. The `timeout` parameter **always** goes inside the command.

```js
// 🚨 DOES NOT WORK
cy.get('.selector').should('be.visible', { timeout: 1000 })
// ✅ THE CORRECT WAY
cy.get('.selector', { timeout: 1000 }).should('be.visible')
```

Remember, you are retrying the command with attached assertions, not just the assertions!

so your `cy.url({ timeout ... })` is correct, but you should change the `cy.xpath` use. If the `cy.xpath` or the `cy.url` do not work, please open an issue with a reproducible example showing that the custom timeouts are NOT applied correctly.
